docs/docs/graphql-api.md

```diff
@@ -0,0 +1,242 @@
+---
+title: GraphQL API
+---
+
+import { GraphqlApiQuery } from "../../www/src/components/api-reference/doc-static-queries"
+import APIReference from "../../www/src/components/api-reference"
+
+A great advantage of Gatsby is a built-in data layer that combines data from any and all data sources you link into. Data is collected at [build time](/docs/glossary#build) and automatically assembled into a [schema](/docs/glossary#schema) that defines how data can be queried throughout components and pages in your site.
+
+This doc serves as a reference for GraphQL features built into Gatsby, including methods for querying and sourcing data, and customizing GraphQL for your site's needs.
+
+## Getting Started with GraphQL
+
+GraphQL doesn't need to be installed in a new Gatsby site, it is already included. A schema is inferred and created when you run `gatsby develop` or `gatsby build` and can be [explored](/docs/running-queries-with-graphiql/) at `http://localhost:8000/___graphql` when a Gatsby site successfully compiles.
+
+## Sourcing Data
+
+Data needs to be [sourced](/docs/content-and-data/) — or added to the GraphQL schema — to be queried and pulled into pages using GraphQL. Gatsby uses [source plugins](/plugins/?=gatsby-source) to pull in, or source, data.
+
+**Note**: GraphQL isn't necessary, you can still [use Gatsby without GraphQL](/docs/using-gatsby-without-graphql/)
+
+Sourcing data with an existing plugin requires installing necessary packages and adding the plugin to the plugins array in the `gatsby-config` with any necessary configurations. To [source data from the filesystem](/docs/sourcing-from-the-filesystem/):
+
+```shell
+npm install --save gatsby-source-filesystem
+```
+
+```js:title=gatsby-config.js
+module.exports = {
+  siteMetadata: {
+    title: `Your Site Name`,
+  },
+  plugins: [
+    // highlight-start
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        name: `src`,
+        path: `${__dirname}/src/`,
+      },
+    },
+    // highlight-end
+  ],
+}
+```
+
+You can also [create custom plugins](/docs/creating-plugins/) to fit your own use cases and pull in data however you want.
+
+## Querying for Data
+
+Data can be queried inside pages, components, or the `gatsby-node.js` file, using one of these options:
+
+- `pageQuery`
+- `StaticQuery`
+- `useStaticQuery`
+
+**Note**: Because of how Gatsby processes GraphQL queries, you can't mix page queries and static queries in the same file. You also can't have multiple page queries or static queries in one file.
+
+### `pageQuery`
+
+You can have one page query per page. A page query can take GraphQL arguments for variables in your queries. A [page is made in Gatsby](/docs/page-creation/) by any React component in the `src/pages` folder, or by calling the `createPage` action and using a component in the `createPage` options -- meaning a `pageQuery` won't work in any component, only components that meet this criteria.
```

Comment:
```suggestion
`pageQuery` is a built-in component that retrieves information from the data layer in Gatsby pages. You can have one page query per page. It can take GraphQL arguments for variables in your queries.

A [page is made in Gatsby](/docs/page-creation/) by any React component in the `src/pages` folder, or by calling the `createPage` action and using a component in the `createPage` options -- meaning a `pageQuery` won't work in any component, only components that meet this criteria.
```